text-kit/autogen.kit.d/base.yml

```diff
@@ -25,6 +29,81 @@ dev_github:
               cmake_src_configure
             }
 
+    - qpdf:
+        github:
+          user: qpdf
+        transform:
+          - kind: string
+            match: 'v '
+            replace: ''
+          - kind: string
+            match: 'release-qpdf-'
+            replace: ''
+        assets:
+          - name: '{{ .Values.pn }}-{{ .Values.version }}.tar.gz'
+            matcher: '{{ .Values.pn }}-{{ .Values.version }}.tar.gz'
+            url: 'https://github.com/qpdf/qpdf/releases/download/v{{ .Values.version }}/{{ .Values.pn }}-{{ .Values.version }}.tar.gz'
+          - name: '{{ .Values.pn }}-{{ .Values.version }}-doc.zip'
+            url: 'https://github.com/qpdf/qpdf/releases/download/v{{ .Values.version }}/{{ .Values.pn }}-{{ .Values.version }}-doc.zip'
+            final_name: '{{ .Values.pn }}-{{ .Values.version }}-doc.zip'
+            unpack: doc
+        vars:
+          license: '|| ( Apache-2.0 Artistic-2 )'
+          slot: '0/$(ver_cut 1)'
```

Comment:
Using helm template you can directly elaborate the first number of the version as macro. I added example in gnome IIRC.
Btw, IMHO, i don't think that it makes sense continue to follow Gentoo with subslotting. Probably you can just drop the slot attribute and use the default value with zero.